apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs

```diff
@@ -59,6 +60,21 @@ public BindingTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHel
                     while (await reader.ReadAsync())
                     {
                         valueRead = reader.GetFieldValue<T>(0);
+
+                        //a few extra tests for certain built in types
+                        if (typeof(T) == typeof(bool)) { Assert.Equal(reader.GetBoolean(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(byte)) { Assert.Equal(reader.GetByte(0).ToString(), valueRead.ToString()); }
+                        // we are disabling culture checks since we do both the same way (and valueRead is a T and cannot be fed culture).
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(decimal)) { Assert.Equal(reader.GetDecimal(0).ToString(), valueRead.ToString()); }
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(double)) { Assert.Equal(reader.GetDouble(0).ToString(), valueRead.ToString()); }
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(float)) { Assert.Equal(reader.GetFloat(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int16)) { Assert.Equal(reader.GetInt16(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int32)) { Assert.Equal(reader.GetInt32(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int64)) { Assert.Equal(reader.GetInt64(0).ToString(), valueRead.ToString()); }
```

Comment:
because at that time the spannerdatareader is shut down and passing around the datareader defeats the purpose of having a good helper function.
There are a finite number of type specific reader calls, so I don't think its a big deal to conditionally test them based on the inputs.